### ISSUE TITLE
Cleanup runners with force delete

### ISSUE THREAD

**Issue by maigl** [cloudbase/garm#50]
I somehow ended up with 3 Runners stuck in state "Creating / Idle". Unfortunately, I don't know how this happened. 
But I have no way to remove those runners. Even a force delete would not remove them:

```
garm-cli runner delete -f  garm-3b02ea89-ae22-4b2e-80b2-e91694ef9c1c
Error: sending request: error in API call: runner must be in "running" or "error" state
```

Of course, this is intended behavior, according to the code. 

How do I clean up those runners? 
Can we remove the requirement: `must be in "running" or "error" state"` ? 

<sub>Michael Kuhnt <michael.kuhnt@mercedes-benz.com> Mercedes-Benz Tech Innovation GmbH ([ProviderInformation](https://github.com/mercedes-benz/daimler-foss/blob/master/PROVIDER_INFORMATION.md))
</sub>

**Comment by gabriel-samfira** [comment]
As a quick hack, I think restarting garm should clean those up. Will look into removing that requirement.

**Comment by gabriel-samfira** [comment]
This one has been bugging me a bit. A ```status``` of ```creating``` means the provider is in the process of spinning up a new instance for the runner. Once the ```CreateInstance()``` function of the provider returns, the ```status``` should transition to ```running``` if the operation succeeded or ```error``` if it failed.

What is interesting is tha the ```runner_status``` is set to ```idle```. That status should be ```pending``` if the instance is in ```creating```. If it transitioned to ```idle`` it means that the instance was spun up, it installed the runner and phoned home telling us it finished installing the runner.

This should only happen if ```garm``` was killed/restarted while it was creating the instance, and before ```CreateInstance()``` had a chance to return. The instance was still created in the provider with the appropriate user data, so it had all it needed to finish bootstrapping.

Do you know if garm was restarted or if it was killed?

At the very least, we should check all instances on startup. If we find an instance in ```creating``` when ```garm``` starts up, we should consider it to be in error, and clean it up.

> Can we remove the requirement: `must be in "running" or "error" state"` ?

The reason why we have that check, is to avoid race conditions. Force deleting a runner does the following things:

  * Check if the runner is already registered and ```idle``` in github
    * If the runner is registered and idle, remove it. Runners that are currently executing a job cannot be deleted
  * Mark the instance as ```pending_delete``` and wait for the consolidation loop to reap it

If an instance were in ```creating``` when we decided to force delete it, we would mark it as ```pending_delete```, but that status would be overwritten once ```CreateInstance()``` would return, setting the instance back to ```running```.

To make this work, we need a mechanism to send a chancel to ```CreateInstance()``` and trigger a cleanup of any created resources. For the time being, we should make sure garm just cleans up those instances on startup.

**Comment by gabriel-samfira** [comment]
I know this is not a real fix, but garm should clean up those runners if they fail to register within the bootstrap timeout set on the pool. If they do register, they should be cleaned up once they are used by a workflow.

I also added a ```Wait()``` on the pool loops when garm is shutting down (this should have been added from the start), so garm will wait for 60 seconds if a loop is busy creating an instance for example. This way we should at least avoid situations where garm bails before the provider returns.

Some consistency checks on startup would be better. When I get some free cycles, I'll look into adding that.
